src/client/client.ts

```diff
@@ -148,6 +222,55 @@ export default class CTProtoClient<AuthRequestPayload, AuthResponsePayload, ApiR
     this.init();
   }
 
+  /**
+   * This method sends file requests
+   *
+   * @param type - available type of requests
+   * @param file - file to send
+   * @param payload - available request payload
+   */
+  public async sendFile(
+    type: ApiUploadRequest['type'],
+    file: Buffer,
+    payload: ApiUploadRequest['payload']
+
+  ): Promise<ApiResponse['payload']> {
+    return new Promise( resolve => {
+      const callback = (response: ApiResponse['payload']): void => {
+        resolve(response);
+      };
+
+      const fileId = MessageFactory.createFileId();
+
+      /**
+       * Calculate number of chunks
+       */
+      const chunks = this.calculateChunksNumber(file);
+
+      /**
+       * Create new uploading file to save data for file uploading
+       */
+      this.uploadingFiles.push({
+        id: fileId,
+        fileData: file,
+        cb: callback,
+        resendTimes: 0,
+      });
+
+      const uploadingFile = this.getUploadingFileById(fileId);
```

Comment:
looks line it has no sence since you just have this object in the prvious statement.

```js
const uploadingFile = {
  id: fileId,
  fileData: file,
  cb: callback,
  resendTimes: 0,
}

this.uploadingFiles.push(uploadingFile)


// searching is not needed

```